app/src/UI/Layout/OverlayLayout/Overlay.tsx

```diff
@@ -1,27 +1,93 @@
+import { AppRoutes } from 'UI/Layout/Routes/AppRoutes';
+import AlertsContainer from 'UI/Layout/AlertsContainer/AlertsContainer';
+import UserInputModalController from 'UI/Reusable/UserInputModals/UserInputModalController';
+import { WebHeader } from 'UI/Layout/OverlayLayout/Header/Web/WebHeader';
+import { MobileOnly } from 'UI/Reusable/Predicates/MobileOnly';
+import { WebOnly } from 'UI/Reusable/Predicates/WebOnly';
+import MobileHeader from 'UI/Layout/OverlayLayout/Header/Mobile/MobileHeader';
+import { Footer } from 'UI/Layout/OverlayLayout/Footer/Footer';
+import NewRecordDialog from 'UI/Features/Records/NewRecordDialog';
+import { OfflineDataSyncDialog } from 'UI/Features/OfflineDataSync/OfflineDataSyncDialog';
+import { OfflineUserMenu } from 'UI/Features/OfflineUserMenu/OfflineUserMenu';
+import CustomizeLayerMenu from 'UI/Features/LegacyMap/Controls/CustomizeLayerDialog';
+import Spinner from 'UI/Reusable/Spinner/Spinner';
+import { DynamicMapComponent } from 'UI/Layout/DynamicMapComponent/DynamicMapComponent';
+import { LayoutMode } from 'UI/Layout/Routes/PrimaryNavigation';
 
-const Overlay = (props) => {
-  const ref = useRef(0);
-  ref.current += 1;
-  if (RENDER_DEBUG) {
-    console.log('%cOverlay render:' + ref.current.toString(), 'color: yellow');
-  }
+const Overlay = () => {
+  const [panelOpen, setPanelOpen] = React.useState(true);
+  const [fullScreen, setFullScreen] = React.useState(false);
 
-  const panelOpen = useSelector((state) => state.AppMode.panelOpen);
-  const fullScreen = useSelector((state) => state.AppMode.panelFullScreen);
+  const layoutMode = useSelector((state) => state.AppMode.layout.viewLayout);
+
+  useEffect(() => {
+    switch (layoutMode) {
+      case LayoutMode.MAP_EXCLUSIVE:
+        setPanelOpen(false);
+        setFullScreen(false);
+        break;
+      case LayoutMode.MAP_HIDDEN:
+        setPanelOpen(true);
+        setFullScreen(true);
+        break;
+      case LayoutMode.MAP_FOCUSED:
+      default:
+        setPanelOpen(true);
+        setFullScreen(false);
+        break;
+    }
+  }, [layoutMode]);
 
   return (
-    <div
-      id="overlaydiv"
-      className={`map__overlay ${panelOpen && !fullScreen ? 'map__overlay--show' : ''} ${
-        panelOpen && fullScreen ? 'map__overlay--show-fullscreen' : ''
-      }`}
-    >
-      <div className={`mapOverlayContents `}>{props.children}</div>
-    </div>
+    <>
+      <AlertsContainer />
+      <UserInputModalController />
+
+      <MobileOnly>
+        <MobileHeader />
+      </MobileOnly>
+
+      <WebOnly>
+        <WebHeader />
+      </WebOnly>
+
+      <Suspense fallback={<Spinner />}>
+        <DynamicMapComponent />
+      </Suspense>
+
+      <div id="overlay-anchor">
+        <div
+          id="overlaydiv"
+          className={`map__overlay ${panelOpen && !fullScreen ? 'map__overlay--show' : ''} ${
+            panelOpen && fullScreen ? 'map__overlay--show-fullscreen' : ''
+          }`}
+        >
```

Comment:
```suggestion
        <div
          id="overlaydiv"
          className={`map__overlay ${(() => {
            if (panelOpen) {
              return fullScreen ? 'map__overlay--show-fullscreen' : 'map__overlay--show';
            }
            return '';
          })()}`}
        >
```

This is a little more intuitive to read than the nested Ternary's with blank returns.
Added a screenshot since GitHub mangles the suggestion

![image](https://github.com/user-attachments/assets/8f6931c6-9f79-4027-93ab-7fd8112d7bd6)